### ISSUE TITLE
Terminating w/uncaught exception of type std::out_of_rante: vector

### ISSUE THREAD

**Issue by SuiteResources** [lvinken/Enc2MusicXML#1]
I'm trying to help out a friend who has some very old Encore files and wants to export to MusicMXL. I've made a couple of tweaks to your code after downloading your repo, such as adding UNKNOWN3, UNKNOWN4, and UNKNOWN5 to encfile.cpp because I was getting "unsupported elemType 13" debugging output and "unsupported elemType 15".

That helped get me through the reading of the .enc file, but now on writing the XML, I get output for about 47 measures, then issues:

xxx_voice_timing measureNr 55
xxx_voice_timing voice 0
"xxx_voice_timing" staff 0 voice 0 tick 0 x_offset 0 note value 3 dots 0 actual 0 normal 0 calcdur 240 nrm 0 0 pbdur 192 nexttick 240
"xxx_voice_timing" staff 0 voice 0 tick 240 x_offset 0 rest value 3 dots 0 calcdur 240 nexttick 480
libc++abi.dylib: terminating with uncaught exception of type std::out_of_range: vector

Any ideas?

I see this in the mxmlconverter.cpp file:

int encKeyToFifths(unsigned int key)
{
    std::vector<int> v =
    {
        //   c   f  bf  ef  af  df  gf  cf  g   d   a   e   b  fs  cs
        /**/ 0, -1, -2, -3, -4, -5, -6, -7, 1,  2,  3,  4,  5,  6,  7
    };
    return v.at(key);
}

Perhaps the issue is here somewhere.

**Comment by lvinken** [comment]
No, I'd need to see more context to understand what is happening exactly.
Do you have a stack trace available ?

**Comment by SuiteResources** [comment]
I only have the debug output.

Here is another place in the mxmlconverter.cpp file where a vector is used.

Line 427:

//---------------------------------------------------------
// createMultiLineString - convert a vector of strings
// to a single newline separated string
//---------------------------------------------------------

static QString createMultiLineString(const std::vector<QString> strVec)
{
    QString res;
    if (strVec.size() > 0) {
        res += strVec.at(0);
        for (size_t i = 1; i < strVec.size(); ++i) {
            if (!strVec.at(i).isEmpty()) {
                res += "\n";
                res += strVec.at(i);
            }
        }
    }
    return res;
}

**Comment by lvinken** [comment]
I'm afraid that does not help. I'd be willing to see if I can find the
issue myself, but that would require access to the enc file (and
preferably also an image of the score).

**Comment by travelalphaplus** [comment]
I think I have  the same issue:

```
Enc2Musicxml/build-Enc2MusicXML-Desktop_Qt_5_12_12_clang_64bit-Debug/src/Enc2MusicXML -m ChuaChanNuoiToi_dt.enc >ChuaChanNuoiToi_dt.musicxml ...
2022-07-31 22:15:34.590674+0700 Enc2MusicXML[8516:3744152] processing file "ChuaChanNuoiToi_dt.enc"
2022-07-31 22:15:34.590836+0700 Enc2MusicXML[8516:3744152] filepos "0" magic "ZBOT"
2022-07-31 22:15:34.590853+0700 Enc2MusicXML[8516:3744152] m_magic "ZBOT"
2022-07-31 22:15:34.590866+0700 Enc2MusicXML[8516:3744152] m_magic "ZBOT" is incorrect
2022-07-31 22:15:34.590893+0700 Enc2MusicXML[8516:3744152] header m_magic "" m_chuMagio 0 m_chuVersio 0 m_nekon1 0 m_fiksa1 0 m_lineCount 0 m_pageCount 0 m_instrumentCount 0 m_staffPerSystem 0 m_measureCount 0
2022-07-31 22:15:39.350250+0700 Enc2MusicXML[8516:3744152] filepos "3c01" magic ""
2022-07-31 22:16:03.936799+0700 Enc2MusicXML[8516:3744152] next id "" var_size 0
libc++abi.dylib: terminating with uncaught exception of type std::out_of_range: vector
```

[ChuaChanNuoiToi_dt.enc.zip](https://github.com/lvinken/Enc2MusicXML/files/9229170/ChuaChanNuoiToi_dt.enc.zip)

**Comment by travelalphaplus** [comment]
I think I have the same issue:

```
Enc2Musicxml/build-Enc2MusicXML-Desktop_Qt_5_12_12_clang_64bit-Debug/src/Enc2MusicXML -m ChuaChanNuoiToi_dt.enc >ChuaChanNuoiToi_dt.musicxml ...
2022-07-31 22:15:34.590674+0700 Enc2MusicXML[8516:3744152] processing file "ChuaChanNuoiToi_dt.enc"
2022-07-31 22:15:34.590836+0700 Enc2MusicXML[8516:3744152] filepos "0" magic "ZBOT"
2022-07-31 22:15:34.590853+0700 Enc2MusicXML[8516:3744152] m_magic "ZBOT"
2022-07-31 22:15:34.590866+0700 Enc2MusicXML[8516:3744152] m_magic "ZBOT" is incorrect
2022-07-31 22:15:34.590893+0700 Enc2MusicXML[8516:3744152] header m_magic "" m_chuMagio 0 m_chuVersio 0 m_nekon1 0 m_fiksa1 0 m_lineCount 0 m_pageCount 0 m_instrumentCount 0 m_staffPerSystem 0 m_measureCount 0
2022-07-31 22:15:39.350250+0700 Enc2MusicXML[8516:3744152] filepos "3c01" magic ""
2022-07-31 22:16:03.936799+0700 Enc2MusicXML[8516:3744152] next id "" var_size 0
libc++abi.dylib: terminating with uncaught exception of type std::out_of_range: vector
```
[ChuaChanNuoiToi_dt.enc.zip](https://github.com/lvinken/Enc2MusicXML/files/9229236/ChuaChanNuoiToi_dt.enc.zip)
